facebook_connector.py

```diff
@@ -0,0 +1,101 @@
+"""
+
+Author: Nigel Schuster <nigel schusters at gmail dot com>
+
+This module provides a simple abstraction for accessing the Facebook chat data
+
+"""
+
+from __future__ import print_function
+
+from bs4 import BeautifulSoup
+from fbchat_archive_parser.parser import MessageHtmlParser
+import fnmatch
+import os
+import pandas as pd
+import re
+import warnings
+
+_messages_file = None
+_fb_id_pattern = re.compile(r"(\d+)@facebook\.com")
+
+
+def initialize(dump_location="."):
+    """
+        Initializes the access to the Facebook messages file by asserting its location
+
+        Args:
+            dump_location: path to the messages.htm file
+    """
+    global _messages_file
+    fb_message_filename = "messages.htm"
+    _messages_file = os.path.join(dump_location, fb_message_filename)
+    if not os.path.isfile(_messages_file):
+        print("Please specify the location of the file in the function call".format(
+            os.path.abspath(_messages_file)))
+        _messages_file = None
+
+
+def resolve_user_id(user_id):
+    """
+    Tries to map the identifier facebook provides to the name of the user.
+
+    Args:
+        user_id: identifier string that is provided in the messages file
+
+    Returns:
+        The name of the user if it is able to find it, otherwise the input
+    """
+    return user_id
+
+
+def get_cleaned_fully_merged_messages(strip_html_content=True,
+                                      resolve_fb_id=False):
+    """
+    Parses the messages file to create dataframes that contain the messages and their senders respectively
+
+    Args:
+        strip_html_content: The messages.htm file might contain some html tags in messages. This option will remove all html markup.
+        resolve_fb_id: The messages.htm file does always print Facebook names, but sometimes ids instead. This will atempt to resolve the issue.
+
+    Returns:
+        a dataframe that contained all messages with info about their senders
+    """
+    if _messages_file is None:
+        print("Please initialize the module first.")
+        return
+    chats = MessageHtmlParser(path=_messages_file).parse()
+    me = chats.user
+    addresses = set()
+    messages = []
+    warnings.filterwarnings("ignore", category=UserWarning, module='bs4')
```

Comment:
We're using BS4 to remove possible html content from messages. Not all messages contain HTML. If they don't, then BS4 throws a warning. Had to add this, otherwise there would be an endless string of these warnings 😄 
The other workaround would be to wrap every message into a <div> before parsing, but that's not better imho.